### PR TITLE
Space laser

### PR DESCRIPTION
**Pull request by osu-murphcol**

### CODE REVIEW

**Comment by osu-murphcol** [comment]
fixed conflicts

File: src/main/java/cs361/battleships/models/Result.java [hunk 1]
```diff
@@ -7,32 +7,43 @@
...
 	public Square getLocation() {
-		return location;
+		return this.location;
+	}
+
+	public boolean getSubmerged() {
+		return this.submerged;
+	}
+
+	public void setSubmerged() {
```
Comment (evanmhm) [inline src/main/java/cs361/battleships/models/Result.java]: This should be like 
setSubmerged(boolean val) {
    this.submerged = val;
}

File: src/main/java/cs361/battleships/models/Board.java [hunk 2]
```diff
@@ -50,42 +49,71 @@ public boolean placeShip(Ship ship, int x, char y, boolean isVertical) {
...
+
+		// Attack surface ships
+		Result attackResult = attack(ships, new Square(x, y) );
 		attacks.add(attackResult);
+
+		// Attack subsurface ships if laser has been earned
+		if ( laserEarned ) {
+
+			Result subAttackResult = attack( submarines, new Square(x, y) );
+			subAttackResult.setSubmerged();
```
Comment (evanmhm) [inline src/main/java/cs361/battleships/models/Board.java]: this should turn into 
subAttackResult.setSubmerged(true);

**Review by evanmhm — CHANGES_REQUESTED** [review]
Looks good besides this little change

File: src/main/java/cs361/battleships/models/Result.java (hunk 1 above)
Comment (osu-murphcol) [reply src/main/java/cs361/battleships/models/Result.java]: Great, we can make those changes!

File: src/main/java/cs361/battleships/models/Board.java (hunk 2 above)
Comment (osu-murphcol) [reply src/main/java/cs361/battleships/models/Board.java]: I agree!

**Pull request merged**